### PR TITLE
feat: support `largelist` in `array_to_string`

### PR DESCRIPTION
**Pull request by Weijun-H**
## Which issue does this PR close?

Parts. #8185

### CODE REVIEW

File: datafusion/physical-expr/src/array_expressions.rs [hunk 1]
```diff
@@ -2054,9 +2067,34 @@ pub fn array_to_string(args: &[ArrayRef]) -> Result<ArrayRef> {
...
     let mut res: Vec<Option<String>> = Vec::new();
 
-    match arr.data_type() {
-        DataType::List(_) | DataType::LargeList(_) | DataType::FixedSizeList(_, _) => {
-            let list_array = arr.as_list::<i32>();
+    let arr_type = arr.data_type();
+    match arr_type {
+        DataType::List(_) | DataType::FixedSizeList(_, _) => {
+            let list_array = as_list_array(&arr)?;
+            for (arr, &delimiter) in list_array.iter().zip(delimiters.iter()) {
```
Comment (jayzhan211) [inline datafusion/physical-expr/src/array_expressions.rs:2074]: Maybe create a function with OffsetSizeTrait to avoid duplicate code?

Comment (jayzhan211) [reply datafusion/physical-expr/src/array_expressions.rs:2074]: Like other function that cast with `as_generic_list_array`

**Review by jayzhan211 — CHANGES_REQUESTED** [review]

**Review by jayzhan211 — APPROVED** [review]
LGTM!

**Review by alamb — APPROVED** [review]
Thank you @Weijun-H  and @jayzhan211

**Pull request merged**